## FINANCE &ACCOUNTING ANALYTICS


In [6]:
#IMPORTING LIBARIES 
import pandas as pd 
import matplotlib.pyplot as plt 
import numpy as np 

In [7]:
#READ DATAFRAME 
df=pd.read_csv('RejectStatsA Ready.csv')

In [8]:
#FIRST 5 ROWS 
df.head()

Amount Requested Application Date                        Loan Title  \
0            1000.0        26/5/2007  Wedding Covered but No Honeymoon   
1            1000.0        26/5/2007                Consolidating Debt   
2           11000.0        27/5/2007       Want to consolidate my debt   
3            6000.0        27/5/2007                           waksman   
4            1500.0        27/5/2007                            mdrigo   

   Risk_Score Debt-To-Income Ratio Zip Code State Employment Length  \
0       693.0                  10%    481xx    NM           4 years   
1       703.0                  10%    010xx    MA          < 1 year   
2       715.0                  10%    212xx    MD            1 year   
3       698.0               38.64%    017xx    MA          < 1 year   
4       509.0                9.43%    209xx    MD          < 1 year   

   Policy Code  
0            0  
1            0  
2            0  
3            0  
4            0

In [145]:
#conditionsFOR RISK SCORE 
conditions = [
    (df['Risk_Score'] >= 800) &(df['Risk_Score']<=850),
    (df['Risk_Score'] >= 750) & (df['Risk_Score'] <=799),
    (df['Risk_Score'] >= 700) & (df['Risk_Score'] <= 749),
    (df['Risk_Score'] >= 650) &(df['Risk_Score']<= 699),
    (df['Risk_Score'] >=600)  &(df['Risk_Score']<=649),
    (df['Risk_Score'] >=300)  &(df['Risk_Score']<=599)]

In [138]:
# values 
values=['excellent', 'very good', 'good', 'fair','poor','very bad']

In [139]:
cond=[
    (df['Debt-To-Income Ratio']> '20%'),
    (df['Debt-To-Income Ratio']>='10%')&(df['Debt-To-Income Ratio']<='20%'),
    (df['Debt-To-Income Ratio']<'10%')
]

In [140]:
val=['high','medium','low']

In [141]:
df['DTl']=np.select(cond,val)

In [142]:
## ADD NEW COLUMN BASED ON CON,VAL USING NP.SELECT 
df['risk_category'] = np.select(conditions, values)


In [143]:
#CHECKING 
df['risk_category'].value_counts()

0                      266944
fair                   230626
very bad               191377
poor                   174060
good                   106058
not possible to cal     52627
very good               23502
excellent                3381
Name: risk_category, dtype: int64

In [111]:
# filter data only to CA
ca=df[df['State']=='CA']

In [112]:
#ONLY CA DATAFRAME 
ca.head()


Amount Requested Application Date              Loan Title  Risk_Score  \
32             7000.0        30/5/2007          big fat lender       638.0   
38             6000.0        30/5/2007  Good Heart Bad Credit!       463.0   
216            2000.0        19/6/2007                 Jlb7557       529.0   
228           10000.0        21/6/2007      Loan Consolidation       635.0   
396            9825.0         6/7/2007                acordars       576.0   

    Debt-To-Income Ratio Zip Code State Employment Length  Policy Code  \
32                17.73%    941xx    CA          < 1 year            0   
38                12.34%    953xx    CA            1 year            0   
216               27.01%    925xx    CA            1 year            0   
228               13.35%    926xx    CA            1 year            0   
396               14.28%    926xx    CA           2 years            0   

    risk_category  row_num     DTl  
32           poor       32  medium  
38       very bad       38  medium  
216      very bad      216    high  
228          poor      228  medium  
396      very bad      396  medium

In [113]:
df['row_num'] = np.arange(len(df))

#### THE NUMEBR OF REJECTED APLICATIONS FOR THE STATE OF CA

In [114]:
ca['State'].value_counts()

CA    133333
Name: State, dtype: int64

#### Count the loans by state and see what percentage of the rejected loans came from California

In [115]:
# pivot table
table = pd.pivot_table(df, values ='row_num',
                         columns =['State'], aggfunc = 'count')
# transpose function 
# apply lambda function to convert to % of propotion 
table.transpose().apply(lambda x:100 * x / float(x.sum())).round(2).head()


row_num
State         
AK        0.26
AL        1.73
AR        1.20
AZ        2.12
CA       12.72

####  ca have 12.72% of rejected loans accross usa # need further investigation 

In [116]:
# CA number of rejected loans ! 
ca.groupby('risk_category').State.count()

risk_category
0            31445
excellent      599
fair         32452
good         15407
poor         22870
very bad     26803
very good     3757
Name: State, dtype: int64

##### IN CA THERE ATE 599 REJECTED APPLIACTIONS DESPITE THEY HAVE EXCLIENT RISK !

#### Determine the number of rejected loans requested by thoserated as having an excellent credit score

In [117]:

table = pd.pivot_table(df, values ='row_num',
                         columns =['DTl','risk_category'], aggfunc = 'count')
table.transpose()


row_num
DTl    risk_category         
high   0               148566
       excellent         1521
       fair            155668
       good             74141
       poor            102090
       very bad        105374
       very good        13931
low    0                56390
       excellent         1141
       fair             14548
       good              8212
       poor             15264
       very bad         79252
       very good         3883
medium 0                62101
       excellent          719
       fair             60410
       good             23705
       poor             56706
       very bad         59265
       very good         5688